src/App.tsx

```diff
@@ -2,14 +2,26 @@ import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
 import Header from "./headerr/Header";
 import Home from "./pages/Home";
 import Profile from "./pages/Profile";
+import RouteGuard from "./components/RouteGuard";
+import NFTDetailsPage from "./pages/NFTDetailsPage";
 
 function App() {
   return (
     <Router>
       <Header />
       <Routes>
         <Route path="/" element={<Home />} />
-        <Route path="/profile" element={<Profile />} />
+        <Route
+          path="/profile"
+          element={
+            <>
+              <RouteGuard />
+              <Profile />
```

Comment:
this is not good, profile page still flickers when i try to visit it

Reply:
i change that, now when user is not connected on Metamask he cant see link to Profile page at all